### PR TITLE
[torch][repeat_interleave] remove stream syncronization if output size is given

### PR DESCRIPTION
**Pull request by serhaty**
Summary: Same as title.

Test Plan:
Rely on CI signal.

Update unit test to exercise new code path as well.

Differential Revision: D28482927

### CODE REVIEW

File: aten/src/ATen/native/Repeat.h [hunk 1]
```diff
@@ -2,21 +2,32 @@
...
   }
   Tensor repeats_ = repeats.contiguous();
   Tensor cumsum = repeats.cumsum(0);
-  int64_t total = cumsum[-1].item<int64_t>();
+  int64_t total;
+  if (output_size.has_value()) {
+    total = output_size.value();
+  } else {
+    total = cumsum[-1].item<int64_t>();
+    TORCH_CHECK(
```
Comment (serhaty) [inline aten/src/ATen/native/Repeat.h:27]: I was not too happy to move this check here. Other two alternatives I could think of was:
1. Do some kind of async check here?
2. Add another flag called "unsafe", if set to True, skip check.

Please let me know if you have guidance/preference on this.

Comment (ngimel) [reply aten/src/ATen/native/Repeat.h:27]: There's no way to do an async check here, but you can add asserts to the kernel checking that repeats are non-negative and that allocated result size matches the total_size argument

**Review by ngimel — COMMENTED** [review]
Can you also please update documentation in _tensor_docs.py to mention the new argument?

File: torch/_torch_docs.py [hunk 2]
```diff
@@ -10419,6 +10419,9 @@ def merge_dicts(*dicts):
     dim (int, optional): The dimension along which to repeat values.
         By default, use the flattened input array, and return a flat output
         array.
+    output_size (int, optional): Total output size for the given axis
```
Comment (serhaty) [inline torch/_torch_docs.py:10422]: if you have better name recommendation, please let me know!

**Review by ngimel — APPROVED** [review]

**Comment by serhaty** [comment]
hey @BowenBao , I am getting error from onnx tests. Do you know what is proper way of updating them? I am basically adding one more optional parameter to torch.repeat_interleave. cc: @SplitInfinity 

failing test: https://app.circleci.com/pipelines/github/pytorch/pytorch/325381/workflows/7fc24ae2-9075-413f-98fb-e88bb9b22410/jobs/13620131

**Comment by BowenBao** [comment]
hi @serhaty, you will need to update the symbolic func
https://github.com/pytorch/pytorch/blob/034a238bab61ae9c10037a2e1f89bff3c11da22d/torch/onnx/symbolic_opset9.py#L2002
and
https://github.com/pytorch/pytorch/blob/034a238bab61ae9c10037a2e1f89bff3c11da22d/torch/onnx/symbolic_opset11.py#L872

unittest can be updated here

https://github.com/pytorch/pytorch/blob/034a238bab61ae9c10037a2e1f89bff3c11da22d/test/onnx/test_pytorch_onnx_onnxruntime.py#L4035

cc @shubhambhokare1

**Pull request closed without merging**